### ISSUE TITLE
Crash if EZFRD64.dll is present

### ISSUE THREAD

**Issue by gabe56f** [ppy/osu#13634]
**Describe the bug:** Some knock-off chinese controller drivers include the following dll to handle vibration.
https://drive.google.com/file/d/14_VBd4n2LMw2o9tqoqDtJo5gLB7Y2jxq/view?usp=sharing
If this dll is present at `C:\Windows\USB Vibration\*\`, the game will refuse to boot up, and create a crashdump in Local/CrashDumps.

Note that this happens even with gamepad/joystick turned off!

**Screenshots or videos showing encountered issue:** 
https://user-images.githubusercontent.com/22561485/123074641-c76bd000-d417-11eb-9f96-9fd66235046d.mp4

**osu!lazer version:** 2021.623.0

**Logs:** No logs. Only a crashdump

Can be found at %appdata%/../Local/CrashDumps/
https://drive.google.com/file/d/1OXi6jq-ni5dgRFTcaqviZD3wnqe16rIU/view?usp=sharing

**Comment by bdach** [comment]
Not sure crash dumps should be publicly shared like above. And I doubt it's caused by anything we do rather than something in sdl2, but I haven't looked at the dump.

**Comment by gabe56f** [comment]
Sharing crash dumps is risky, with that I agree, but sadly there's nothing else to show. The problem in my opinion is that this happens even with joysticks disabled.

**Comment by peppy** [comment]
Event log should have crash details.

**Comment by gabe56f** [comment]
Give me a second, let me check.

**Comment by gabe56f** [comment]
Indeed, it is there.
https://hastebin.com/oreniyeseq.apache

I have removed the crash dump from my drive.

**Comment by bdach** [comment]
That's a generic access violation error. Not much I can recommend about that on its own other than uninstalling the offending driver which I understand is not an option.

**Comment by peppy** [comment]
Safe to close this as something we can't guard against?

**Comment by bdach** [comment]
I'd say so. I googled around yesterday and found evidence of several other games having the same sort of trouble with that particular driver, so likely an SDL level error or higher as I suggested above.

**Comment by peppy** [comment]
The sad thing is this is likely going to go come up many times after more users start to switch to lazer. @gabe56f as a point of reference, does osu!(stable) also crash in a similar way?

**Comment by gabe56f** [comment]
Stable launches without any issues

**Comment by peppy** [comment]
Will leave this open for now in case other users run into the same thing. Probably not worth investigating further until we know whether this is going to be an issue for more than one user.

**Comment by Squall-Leonhart** [comment]
the dll is incompatible with ASLR binaries following windows updates published in 2016, developers have the option of hot patching it similarly to what dolphin did in https://github.com/dolphin-emu/dolphin/pull/5582/commits/1210c74955f1d680eb639317d33f6324b57ec8d3

debugging results were published to dolphin-emu's forum [here](https://forums.dolphin-emu.org/Thread-dolphin-dev-builds-past-5-0-3977-crash-on-startup-with-my-controller-plugged-in?page=2)

**Comment by peppy** [comment]
Jesus that patch looks painful. I think if we are to apply any local fixes it would probably involve looking for the dll on the system and showing a dialog on startup to remove it else the game won't run.

**Comment by Squall-Leonhart** [comment]
Agreed, i don't recommend circumventing the issue, you can also reject it as performed in https://github.com/7hong13/Aircom/blob/e73a0eae740b8b5aa913940cbf1420b5b156af43/client_pc/AntiHooking/antihookingprotection.cpp

**Comment by peppy** [comment]
Thanks for the useful reference. That looks closer to something we'd be looking at doing.

**Comment by Nasko69** [comment]
> **Describe the bug:** Some knock-off controller drivers include the following dll to handle vibration. https://drive.google.com/file/d/14_VBd4n2LMw2o9tqoqDtJo5gLB7Y2jxq/view?usp=sharing If this dll is present at `C:\Windows\USB Vibration\*\`, the game will refuse to boot up, and create a crashdump in Local/CrashDumps.
> 
> Note that this happens even with gamepad/joystick turned off!
> 
> **Screenshots or videos showing encountered issue:** https://user-images.githubusercontent.com/22561485/123074641-c76bd000-d417-11eb-9f96-9fd66235046d.mp4
> 
> **osu!lazer version:** 2021.623.0
> 
> **Logs:** No logs. Only a crashdump
> 
> Can be found at %appdata%/../Local/CrashDumps/ https://drive.google.com/file/d/1OXi6jq-ni5dgRFTcaqviZD3wnqe16rIU/view?usp=sharing
